test/distributed/_tensor/test_matrix_ops.py

```diff
@@ -24,6 +26,15 @@
 )
 
 
+def tiled_mul(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+    if b.numel() == 1:
+        return a / b.squeeze()
```

Comment:
Is this supposed to be division, or multiplication?

Reply:
Good catch! I'll answer to two comments at once. Indeed, there was some confusion as to where we need to multiply and divide. I tried to fix this, by matching what we're doing in https://github.com/facebookresearch/lingua/pull/65, that is, we divide by the scales before we cast to fp8, and we make scaled_mm multiply back.

The reason we weren't seeing failures in the tests is because the test was only comparing a non-sharded scaled_mm with a sharded one, and in our case they were both incorrect, thus they matched. I fixed this by comparing against an unsharded _high-precision_ matmul instead. This will catch issues in the scaling logic, but it requires much higher tolerances (because fp8 is inherently more inaccurate). Note that this actually simplifies the test logic quite a bit!